File: src/main/java/ExampleRun.java
import data.Repository;
import data.TransientRepository;
import data.model.Footballer;
import exception.NoFplResponseException;
import fpl.score.FixtureCalculator;
import fpl.selection.Selection;

import java.io.IOException;
import java.util.Collections;
import java.util.List;

public class ExampleRun {

    private static final int WEEKS_TO_EVALUATE = 5;
    private static int MANAGER_ID = 2029893;

    public static void main(String[] args) throws IOException {
        long startT = System.currentTimeMillis();

        try {
            Repository repository = new TransientRepository();
            Selection selection = new Selection(repository);
            List<Footballer> footballers = selection.getList(MANAGER_ID);

            FixtureCalculator calculator = new FixtureCalculator(repository);
            calculator.addOppositionForWeeks(selection, WEEKS_TO_EVALUATE);

            Collections.sort(footballers);
            Collections.reverse(footballers);
            for (Footballer footballer : footballers) {
                System.out.println("Player: " + footballer.getWebName() +
                                "| Opponent: " + footballer.getOpponentList().toString() +
                                "| Total: " + footballer.getDifficultyTotal());
            }

        } catch (NoFplResponseException e) {
            System.out.println("FAILED to get required FPL data: " + e.getMessage());
            // Handle failure
        }

        long endT = System.currentTimeMillis();
        System.out.println("Total: " + (endT - startT) + "ms");
    }
}

File: src/main/java/ExampleRun.kt
import data.TransientRepository
import data.model.Footballer
import exception.NoFplResponseException
import fpl.score.FixtureCalculator
import java.io.IOException

object ExampleRun {

    private const val WEEKS_TO_EVALUATE = 5
    private const val MANAGER_ID = 2029893

    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val startTime = System.currentTimeMillis()
        try {
            val repository = TransientRepository()
            val calculator = FixtureCalculator(repository)
            val footballers: MutableList<Footballer> = calculator.getFixturesDifficulty(MANAGER_ID, WEEKS_TO_EVALUATE)

            footballers.sortByDescending { footballer -> footballer.difficultyTotal }
            for (footballer in footballers) {
                val name = footballer.webName
                val opponentString = footballer.opponentList.toString()
                val total = footballer.difficultyTotal
                println("Player: $name | Opponent: $opponentString | Total: $total")
            }
        } catch (e: NoFplResponseException) {
            println("FAILED to get required FPL data: " + e.message)
            // Handle failure
        }

        val endTime = System.currentTimeMillis()
        println("Total: " + (endTime - startTime) + "ms")
    }
}
